barretenberg/cpp/src/barretenberg/ecc/fields/field_conversion.hpp

```diff
@@ -68,75 +43,40 @@ template <typename T> constexpr size_t calc_num_bn254_frs()
-{
-    ASSERT(fr_vec.size() == 1);
-    return static_cast<T>(fr_vec[0]);
-}
-
-bb::fr convert_from_bn254_frs(std::span<const bb::fr> fr_vec, bb::fr* /*unused*/);
-
-grumpkin::fr convert_from_bn254_frs(std::span<const bb::fr> fr_vec, grumpkin::fr* /*unused*/);
-
-curve::BN254::AffineElement convert_from_bn254_frs(std::span<const bb::fr> fr_vec,
-                                                   curve::BN254::AffineElement* /*unused*/);
-
-curve::Grumpkin::AffineElement convert_from_bn254_frs(std::span<const bb::fr> fr_vec,
-                                                      curve::Grumpkin::AffineElement* /*unused*/);
-
-template <size_t N>
-inline std::array<bb::fr, N> convert_from_bn254_frs(std::span<const bb::fr> fr_vec, std::array<bb::fr, N>* /*unused*/)
-{
-    std::array<bb::fr, N> val;
-    for (size_t i = 0; i < N; ++i) {
-        val[i] = fr_vec[i];
-    }
-    return val;
-}
-
-template <size_t N>
-inline std::array<grumpkin::fr, N> convert_from_bn254_frs(std::span<const bb::fr> fr_vec,
-                                                          std::array<grumpkin::fr, N>* /*unused*/)
-{
-    std::array<grumpkin::fr, N> val;
-    for (size_t i = 0; i < N; ++i) {
-        std::vector<bb::fr> fr_vec_tmp{ fr_vec[2 * i],
-                                        fr_vec[2 * i + 1] }; // each pair of consecutive elements is a grumpkin::fr
-        val[i] = convert_from_bn254_frs<grumpkin::fr>(fr_vec_tmp);
-    }
-    return val;
-}
-
-template <size_t N>
-inline Univariate<bb::fr, N> convert_from_bn254_frs(std::span<const bb::fr> fr_vec, Univariate<bb::fr, N>* /*unused*/)
-{
-    Univariate<bb::fr, N> val;
-    for (size_t i = 0; i < N; ++i) {
-        val.evaluations[i] = fr_vec[i];
-    }
-    return val;
-}
-
-template <size_t N>
-inline Univariate<grumpkin::fr, N> convert_from_bn254_frs(std::span<const bb::fr> fr_vec,
-                                                          Univariate<grumpkin::fr, N>* /*unused*/)
+template <typename T> T convert_from_bn254_frs(std::span<const bb::fr> fr_vec)
 {
-    Univariate<grumpkin::fr, N> val;
-    for (size_t i = 0; i < N; ++i) {
-        std::vector<bb::fr> fr_vec_tmp{ fr_vec[2 * i], fr_vec[2 * i + 1] };
-        val.evaluations[i] = convert_from_bn254_frs<grumpkin::fr>(fr_vec_tmp);
+    if constexpr (IsAnyOf<T, bool>) {
+        ASSERT(fr_vec.size() == 1);
+        return fr_vec[0] != 0;
+    } else if constexpr (IsAnyOf<T, uint32_t, bb::fr>) {
+        ASSERT(fr_vec.size() == 1);
+        return static_cast<T>(fr_vec[0]);
+    } else if constexpr (IsAnyOf<T, grumpkin::fr>) {
+        ASSERT(fr_vec.size() == 2);
+        return convert_grumpkin_fr_from_bn254_frs(fr_vec);
+    } else if constexpr (IsAnyOf<T, curve::BN254::AffineElement, curve::Grumpkin::AffineElement>) {
+        using BaseField = typename T::Fq;
+        constexpr size_t BaseFieldScalarSize = calc_num_bn254_frs<BaseField>();
```

Comment:
We've historically used `THIS_CASE` for constexpr values, and `ThisCase` for classes. Could you switch here and elsewhere?

Reply:
done